resistor-color-trio: add exercise (#1551)
exercism/problem-specifications: exercises/resistor-color-trio/description.md

  In this exercise, you are going to create a helpful program so that you don't have to remember the values of the bands. The program will take 3 colors as input, and outputs the correct value, in Ohms.
  The color bands are encoded as follows:

* Black: 0
* Brown: 1
* Red: 2
* Orange: 3
* Yellow: 4
* Green: 5
* Blue: 6
* Violet: 7
* Grey: 8
* White: 9

In `resistor-color duo` you decoded the first two colors. For instance: orange-orange got the main value `33`.
The third color stands for how many zeros need to be added to the main value. The main value plus the zeros gives us a value in Ohms.
For the exercise it doesn't matter what Ohms really are.
For example:

- orange-orange-black would be 33 and no zeros, which becomes 33 Ohms.
- orange-orange-red would be 33 and 2 zeros, which becomes 3300 Ohms.
- orange-orange-orange would be 33 and 3 zeros, which becomes 33000 Ohms.

(If Math is your thing, you may want to think of the zeros as exponents of 10. If Math is not your thing, go with the zeros. It really is the same thing, just in plain English instead of Math lingo.)

This exercise is about translating the colors into a sentence:

> > "The value of this resistor is … Ohms."

Reviewer: There's another unicode character here, `…`

Reviewer: The dots are still there

> > "The value of this resistor is … Ohms."

Reviewer 2: ```suggestion
> "The value of this resistor is ... Ohms."
```

> So an input of `"orange”, "orange”, "black”` should return

Reviewer: Here too: `”`

> > "The value of this resistor is 33 Ohms.”

Reviewer: And here.

> The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say “kilo”, in the same way that we say “kilograms” (kg) and “kilometers” (km). In the same way, we get “kilo-ohms” (kOhms).

Reviewer: For consistency and readability I'd prefer either writing out the unit completely (`kilo Ohms`) or in short form (`kΩ`), but not mixed.

I think you're using the unicode quotation marks here `“”` instead of `""`, is that intended?

Reviewer 2: Thanks Sascha! 
> writing out the unit completely (`kilo Ohms`) or in short form (`kΩ`), but not mixed.

We want to avoid special characters in this one, so let's go with 'kilo Ohms'.

>Using kilograms as example is slightly misleading 
I understand your reasoning, but since it's just about explaining where the 'kilo'-part is coming from in a non-mathy way, and since the base unit is not really relevant for this particular exercise, I'd rather leave it as is. Can you live with that?

> Quotation marks: not intended.

Reviewer: > Can you live with that?

Yes, of course. All of my comments were meant as suggestions that you may or may not consider, not requests for change. I should've made that clearer perhaps. :)

> The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say “kilo”, in the same way that we say “kilograms” (kg) and “kilometers” (km). In the same way, we get “kilo-ohms” (kOhms).

Reviewer 2: ```suggestion
The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say "kilo", in the same way that we say "kilograms" (kg) and "kilometers" (km). In the same way, we get "kilo-ohms" (kilo Ohms).
```

Reviewer 3: I think reversing them here would already be helpful:

```suggestion
The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say "kilo", in the same way that we say "kilometers" (km) and "kilograms" (kg). In the same way, we get "kilo-ohms" (kilo Ohms).
```

I'm not a fan of the double "in the same way".

Perhaps:

```suggestion
The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say "kilo", which results in  "kilo-ohms" (kilo Ohms). In the same way when we have more than a thousand metres we say "kilometers" (km) and when we have more than a thousand grams, we say "kilograms" (kg).
```

Reviewer 2: I see your point, @SleeplessByte, but I'm not convinced by this solution. I'll think of something else.

Reviewer 4: Wouldn’t it be “kilohms” not “Kilo-ohms”? Wolfram Alpha seems think so, and that led me to NIST and https://www.nist.gov/physical-measurement-laboratory/nist-guide-si-chapter-6#623, which indicates prefixes are inseparable from units. Elsewhere it mentions lowercase presentation and that:

> there are three cases in which the final vowel of an SI prefix is commonly omitted: megohm (not megaohm), kilohm (not kiloohm), and hectare (not hectoare)

Although I’ll be honest the only one of those I’m totally sure of is hectare.

Reviewer 3: Yes, hectare is the one I'm also sure about. @F3PiX do as you see fit ❤️ !

> The trick is that when we talk about Ohms, as soon as we get more than a thousand, we say "kilo", in the same way that we say "kilograms" (kg) and "kilometers" (km). In the same way, we get "kilo-ohms" (kilo Ohms).

Reviewer 2: ```suggestion
When we get more than a thousand Ohms, we say "kilo Ohms". That's similar to saying "kilometer" for 1000 meters, and "kilograms" for 1000 grams. 
```
